### DISCUSSION TITLE
Automerge on Github not working

### DISCUSSION THREAD

**Discussion by ggogel** [renovatebot/renovate#26840] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

I want to automerge PRs in a private Github project. Only minor and patch updates. The repository consists of Helm and Kubernetes files. None of the branches is protected. Auto merge is enabled in the project. I don't understand why Renovate is not automerging.

This is my config:

```
{
  "extends": ["config:base"],
  "separateMajorMinor": true,
  "kubernetes": {
    "fileMatch": ["k8s/.+\\.yaml$"]
  },
  "packageRules": [
    {
      "matchUpdateTypes": ["minor", "patch"],
      "matchCurrentVersion": "!/^0/",
      "automerge": true
    }
  ]
}
```

This is a Screenshot of a PR. As you can see it says `Automerge: Enabled` in the PR and at the bottom it says `Merging can be performed automatically`.
<img width="837" alt="image" src="https://github.com/renovatebot/renovate/assets/13143850/674cddf3-cc4c-4674-8ec0-4cc73bad2c0e">

See the full log attached. I found these lines regarding automerge:
```
DEBUG: Checking if we can automerge branch
DEBUG: mergeStatus=no automerge
```
and 

```
DEBUG: PR is not ready for merge (branch status is yellow)
```

But why?

### Logs (if relevant)

<details><summary>Logs</summary>

DEBUG: Using RE2 as regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in /runner/renovate/job_config.json
DEBUG: Detected config in env RENOVATE_CONFIG
DEBUG: File config
DEBUG: CLI config
DEBUG: Env config
DEBUG: Combined config
DEBUG: Enabling forkProcessing while in non-autodiscover mode
DEBUG: Found valid git version: 2.43.0
DEBUG: Setting global hostRules
DEBUG: Adding password authentication for docker.io (hostType=docker) to hostRules
DEBUG: Adding password authentication for hub.docker.com (hostType=docker) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-tags) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-releases) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=pod) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=terraform-provider) to hostRules
DEBUG: Adding token authentication for github.com (hostType=rubygems) to hostRules
DEBUG: Using default github endpoint: https://api.github.com/
DEBUG: Platform config
DEBUG: Using configured gitAuthor (renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>)
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
DEBUG: Using baseDir: /tmp/renovate
DEBUG: Using cacheDir: /tmp/renovate/cache
DEBUG: Using containerbaseDir: /tmp/renovate/cache/containerbase
DEBUG: Redis cache init
DEBUG: Commits limit = null
DEBUG: Setting global hostRules
DEBUG: Adding password authentication for docker.io (hostType=docker) to hostRules
DEBUG: Adding password authentication for hub.docker.com (hostType=docker) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-tags) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-releases) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=pod) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=terraform-provider) to hostRules
DEBUG: Adding token authentication for github.com (hostType=rubygems) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
DEBUG: validatePresets()
DEBUG: Reinitializing hostRules for repo
DEBUG: Clearing hostRules
DEBUG: Adding password authentication for docker.io (hostType=docker) to hostRules
DEBUG: Adding password authentication for hub.docker.com (hostType=docker) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-tags) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github-releases) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=pod) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=terraform-provider) to hostRules
DEBUG: Adding token authentication for github.com (hostType=rubygems) to hostRules
DEBUG: Adding token authentication for api.github.com (hostType=github) to hostRules
INFO: Repository started
DEBUG: Using localDir: /tmp/renovate/repos/github/ggogel/gogel-infra
DEBUG: PackageFiles.clear() - Package files deleted
DEBUG: initRepo("ggogel/gogel-infra")
DEBUG: hostRules: authentication already set for api.github.com
DEBUG: ggogel/gogel-infra default branch = main
DEBUG: Using app token for git init
DEBUG: RepoCacheS3.read() - success
DEBUG: Repository cache is restored from revision 13
DEBUG: Resetting npmrc
DEBUG: Resetting npmrc
DEBUG: checkOnboarding()
DEBUG: isOnboarded()
DEBUG: findPr(renovate/configure, Configure Renovate, !open)
DEBUG: getPrList success
DEBUG: findPr(renovate/configure, chore: Configure Renovate, !open)
DEBUG: Checking cached config file name
DEBUG: Existing config file confirmed
DEBUG: Repository config
DEBUG: Repo is onboarded
DEBUG: migrateAndValidate()
DEBUG: Config migration necessary
DEBUG: Post-massage config
DEBUG: Setting hostRules from config
DEBUG: Found repo ignorePaths
DEBUG: No vulnerability alerts found
DEBUG: No vulnerability alerts found
DEBUG: findIssue(Dependency Dashboard)
DEBUG: Retrieving issueList
DEBUG: Retrieved 1 issues
DEBUG: Found issue 3
DEBUG: No baseBranches
DEBUG: extract()
DEBUG: Cached extract for sha=4b4f8dcdea37ea07418d79379186766782c5df24 is valid and can be used
DEBUG: Deleted cached dep updates
INFO: Dependency extraction complete
DEBUG: hostRules: basic auth for https://index.docker.io
DEBUG: hostRules: applying Basic authentication for hub.docker.com
DEBUG: PackageFiles.add() - Package file saved for base branch
DEBUG: Package releases lookups complete
DEBUG: branchifyUpgrades
DEBUG: detectSemanticCommits()
DEBUG: semanticCommits: returning "disabled" from cache
DEBUG: 3 flattened updates found: wordpress, mariadb, docker.io/bitnami/postgresql
DEBUG: Returning 3 branch(es)
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
DEBUG: detectSemanticCommits()
DEBUG: semanticCommits: returning "disabled" from cache
DEBUG: processRepo()
DEBUG: Processing 3 branches: renovate/docker.io-bitnami-postgresql-16.x, renovate/mariadb-11.x, renovate/wordpress-19.x
DEBUG: Calculating hourly PRs remaining
DEBUG: currentHourStart=2024-01-24T06:00:00.000+00:00
DEBUG: PR hourly limit remaining: 2
DEBUG: Calculating prConcurrentLimit (10)
DEBUG: getBranchPr(renovate/wordpress-19.x)
DEBUG: findPr(renovate/wordpress-19.x, undefined, open)
DEBUG: Found PR #24
DEBUG: getBranchPr(renovate/docker.io-bitnami-postgresql-16.x)
DEBUG: findPr(renovate/docker.io-bitnami-postgresql-16.x, undefined, open)
DEBUG: findPr(renovate/docker.io-bitnami-postgresql-16.x, undefined, closed)
DEBUG: Found PR #11
DEBUG: getBranchPr(renovate/mariadb-11.x)
DEBUG: findPr(renovate/mariadb-11.x, undefined, open)
DEBUG: findPr(renovate/mariadb-11.x, undefined, closed)
DEBUG: Found PR #2
DEBUG: 1 PRs are currently open
DEBUG: PR concurrent limit remaining: 9
DEBUG: Calculated maximum PRs remaining this run: 2
DEBUG: PullRequests limit = 2
DEBUG: Calculating hourly PRs remaining
DEBUG: currentHourStart=2024-01-24T06:00:00.000+00:00
DEBUG: PR hourly limit remaining: 2
DEBUG: Calculating branchConcurrentLimit (10)
DEBUG: 1 already existing branches found: renovate/wordpress-19.x
DEBUG: Branch concurrent limit remaining: 9
DEBUG: Calculated maximum branches remaining this run: 2
DEBUG: Branches limit = 2
DEBUG: syncBranchState()
DEBUG: branch.isUpToDate(): using cached result "true"
DEBUG: getBranchPr(renovate/wordpress-19.x)
DEBUG: findPr(renovate/wordpress-19.x, undefined, open)
DEBUG: Found PR #24
DEBUG: branchExists=true
DEBUG: dependencyDashboardCheck=undefined
DEBUG: PR rebase requested=false
DEBUG: Check for closed PR because recreating closed PRs is disabled.
DEBUG: findPr(renovate/wordpress-19.x, Update wordpress Docker tag to v19.2.1, !open)
DEBUG: prAlreadyExisted=false
DEBUG: Checking if PR has been edited
DEBUG: branch.isModified(): using cached result "false"
DEBUG: Found existing branch PR
DEBUG: Checking schedule(at any time, null)
DEBUG: No schedule defined
DEBUG: Branch already exists
DEBUG: branch.isBehindBase(): using cached result "false"
DEBUG: Branch is up-to-date
DEBUG: isBranchConflicted(main, renovate/wordpress-19.x)
DEBUG: branch.isConflicted(): using cached result "false"
DEBUG: Branch does not need rebasing
DEBUG: Using reuseExistingBranch: true
DEBUG: Checking if we can automerge branch
DEBUG: mergeStatus=no automerge
DEBUG: Ensuring PR
DEBUG: There are 0 errors and 0 warnings
DEBUG: getBranchPr(renovate/wordpress-19.x)
DEBUG: findPr(renovate/wordpress-19.x, undefined, open)
DEBUG: Found PR #24
DEBUG: getPrCache()
DEBUG: Found existing PR
DEBUG: PR cache matches but it has been edited in the past 24hrs, so processing PR
DEBUG: Fetching changelog: https://github.com/bitnami/charts (19.2.0 -> 19.2.1)
DEBUG: Processing existing PR
DEBUG: resolveBranchStatus(branchName=renovate/wordpress-19.x, ignoreTests=false)
DEBUG: getBranchStatus(renovate/wordpress-19.x)
DEBUG: branch status check result
DEBUG: Successful checks are all internal renovate/ checks, so returning "pending" branch status
DEBUG: No check runs found
DEBUG: Branch status yellow
DEBUG: setPrCache()
DEBUG: Pull Request #24 does not need updating
DEBUG: PR is configured for automerge
DEBUG: checking auto-merge
DEBUG: Checking schedule(at any time, null)
DEBUG: No schedule defined
DEBUG: resolveBranchStatus(branchName=renovate/wordpress-19.x, ignoreTests=false)
DEBUG: getBranchStatus(renovate/wordpress-19.x)
DEBUG: branch status check result
DEBUG: Successful checks are all internal renovate/ checks, so returning "pending" branch status
DEBUG: No check runs found
DEBUG: Branch status yellow
DEBUG: PR is not ready for merge (branch status is yellow)
DEBUG: syncBranchState()
DEBUG: branch.isUpToDate(): using cached result "true"
DEBUG: getBranchPr(renovate/docker.io-bitnami-postgresql-16.x)
DEBUG: findPr(renovate/docker.io-bitnami-postgresql-16.x, undefined, open)
DEBUG: findPr(renovate/docker.io-bitnami-postgresql-16.x, undefined, closed)
DEBUG: Found PR #11
DEBUG: branchExists=false
DEBUG: dependencyDashboardCheck=undefined
DEBUG: Check for closed PR because recreating closed PRs is disabled.
DEBUG: findPr(renovate/docker.io-bitnami-postgresql-16.x, Update docker.io/bitnami/postgresql Docker tag to v16, !open)
DEBUG: Found PR #11
DEBUG: Found closed PR with current title
DEBUG: Returning PR from cache
DEBUG: Closed PR already exists. Skipping branch.
DEBUG: syncBranchState()
DEBUG: branch.isUpToDate(): using cached result "true"
DEBUG: getBranchPr(renovate/mariadb-11.x)
DEBUG: findPr(renovate/mariadb-11.x, undefined, open)
DEBUG: findPr(renovate/mariadb-11.x, undefined, closed)
DEBUG: Found PR #2
DEBUG: branchExists=false
DEBUG: dependencyDashboardCheck=undefined
DEBUG: Check for closed PR because recreating closed PRs is disabled.
DEBUG: findPr(renovate/mariadb-11.x, Update mariadb Docker tag to v11, !open)
DEBUG: Found PR #2
DEBUG: Found closed PR with current title
DEBUG: Returning PR from cache
DEBUG: Closed PR already exists. Skipping branch.
DEBUG: getBranchPr(renovate/wordpress-19.x)
DEBUG: findPr(renovate/wordpress-19.x, undefined, open)
DEBUG: Found PR #24
DEBUG: getPrCache()
DEBUG: getPrCache()
DEBUG: getPrCache()
DEBUG: ensureDependencyDashboard()
DEBUG: Ensuring Dependency Dashboard
DEBUG: ensureIssue(Dependency Dashboard)
DEBUG: Issue is open and up to date - nothing to do
DEBUG: validateReconfigureBranch()
DEBUG: No reconfigure branch found
DEBUG: Removing any stale branches
DEBUG: config.repoIsOnboarded=true
DEBUG: Branch lists
DEBUG: remainingBranches=
DEBUG: No branches to clean up
DEBUG: PackageFiles.clear() - Package files deleted
DEBUG: Repo is activated
DEBUG: Branch summary
DEBUG: branches info extended
DEBUG: Renovate repository PR statistics
DEBUG: Repository result: done, status: activated, enabled: true, onboarded: true
DEBUG: Repository timing splits (milliseconds)
DEBUG: Package cache statistics
DEBUG: http statistics
DEBUG: Package lookup durations
DEBUG: dns cache
INFO: Repository finished
DEBUG: Renovate exiting

</details>

**Comment by HonkingGoose** [comment]
By the way `config:base` is now `config:recommended`. [^1] You should update your config to use the new term. That way you can easily refer to the Renovate docs later.

Did you read the [Renovate docs, key concepts, automerge](https://docs.renovatebot.com/key-concepts/automerge/) page? Maybe that has information to fix your problem.

[^1]: [Renovate docs, `config:recommended` preset](https://docs.renovatebot.com/presets-config/#configrecommended)

**Reply by ggogel** [reply]
Hi, thank you for the response.

I'll change it to `config:recommended`.

Yes, I read the docs. That's where I got the part of the config from.

**Reply by ggogel** [reply]
After setting `config:recommended` the behavior was unchanged.

I found this old thread https://github.com/renovatebot/config-help/issues/784

There it is recommended to set `"requiredStatusChecks": null`. Setting this made the auto merges work but I can't find any reference to this in the docs and it feels kind of like a hack.

**Reply by ggogel** [reply]
Even after manually triggering Renovate through the dependency dashboard again, branch status stays `yellow`. Although Github says the PR is ready for automerge.

**Reply by ggogel** [reply]
By the way, when opening the tab checks on the PR in Github it shows `There are no checks for this commit`. So, maybe the issue is that there are no checks. If no checks are defined, it can't get a successful check. Then the status is evaluated to yellow.

Sadly Renovate does not explain why the status is evaluated to yellow.

**Reply by viceice** [reply, marked as answer]
no, by default renovate only automerge when there is at least one status check and all status checks are successful. you can renovate tell to skip checks, but that can lead to broken target branch.

https://docs.renovatebot.com/configuration-options/#ignoretests

**Reply by ggogel** [reply]
I see, I just edited my answer above when you were replying and came to the same conclusion.

So I'll set this option then since I can't think of a test for this type of project that has just Helm and Kubernetes files.

Thank you for your help.

**Reply by HonkingGoose** [reply]
I always like to have a basic _does my project build or start_ test. Is that suitable for your project?

**Reply by ggogel** [reply]
There is nothing I could build or start since the project only consists of YAML files that describe my production environment. The only thing I can think of is checking if those files are correctly formatted but this is checked by Argo CD anyway.

**Reply by HonkingGoose** [reply]
Oh okay. :smile:

**Reply by CommanderStorm** [reply]
@ggogel Note that tests in helm are possible ^^
https://helm.sh/docs/topics/chart_tests/
